client/crypto.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+
+	commonv1pb "github.com/dapr/go-sdk/dapr/proto/common/v1"
+	runtimev1pb "github.com/dapr/go-sdk/dapr/proto/runtime/v1"
+)
+
+// Encrypt data read from a stream, returning a readable stream that receives the encrypted data.
+// This method returns an error if the initial call fails. Errors performed during the encryption are received by the out stream.
+func (c *GRPCClient) Encrypt(ctx context.Context, in io.Reader, opts EncryptOptions) (io.Reader, error) {
+	// Ensure required options are present
+	// This short-circuits and avoids a call to the runtime
+	if opts.ComponentName == "" {
+		return nil, errors.New("option 'ComponentName' is required")
```

Comment:
Better to make them as const.

Reply:
I am not a fan of making errors as constants:

1. They are only used once or twice. Defining them as constants requires the person reading the code to jump back and forth to read what the constant is. It makes reading the code harder
2. The difference would be if we expected callers to perform `errors.Is` to check if the error is the same. These errors are generally “dev-time” errors so users are not expected to check if the errors match.